src/Lookup/UnresolvedEntityRedirectException.php

```diff
@@ -22,10 +21,11 @@ class UnresolvedRedirectException extends RuntimeException {
 	private $redirectTargetId;
 
 	/**
+	 * @param EntityId $entityId
 	 * @param EntityId $redirectTargetId The ID of the target Entity of the redirect
 	 */
-	public function __construct( EntityId $redirectTargetId ) {
-		parent::__construct( "Unresolved redirect to " . $redirectTargetId->getSerialization() );
+	public function __construct( EntityId $entityId, EntityId $redirectTargetId ) {
+		parent::__construct( $entityId, "Unresolved redirect to " . $redirectTargetId->getSerialization() );
```

Comment:
Subclassing is one thing that can be done. Keep in mind that not all code in Wikibase.git needs this extra info, just some of it. Loose from these concepts not belonging in this component, they should not have been put into UnresolvedRedirectException to begin with.